packages/app/src/systems/SignUp/components/CreatePassword/CreatePassword.tsx

```diff
@@ -60,45 +66,70 @@ export function CreatePassword({
           subtitle="Add a safe password for access your wallet"
         />
         <Stack css={{ width: "100%" }} gap="$4">
-          <Form.Control isInvalid={Boolean(errors.password)}>
-            <Form.Label htmlFor="password">Password</Form.Label>
-            <InputPassword
-              {...register("password")}
-              id="password"
-              name="password"
-              placeholder="Type your password"
-            />
-            {errors.password?.message && (
-              <Form.ErrorMessage>
-                {errors.password?.message as string}
-              </Form.ErrorMessage>
+          <Controller
+            name="password"
+            control={control}
+            render={({ field, fieldState }) => (
+              <Form.Control isInvalid={Boolean(fieldState.error)}>
+                <Form.Label htmlFor="password">Password</Form.Label>
+                <InputPassword
+                  {...field}
+                  id="password"
+                  placeholder="Type your password"
+                />
+                {fieldState.error?.message && (
```

Comment:
This will be a common thing. Can we split this logic into a component?

Reply:
✅